tune/protox/agent/hpo.py

```diff
@@ -465,11 +497,15 @@ def step(self) -> dict[Any, Any]:
                 f"Baseline Metric: {baseline_metric}. Baseline Reward: {baseline_reward}"
             )
             self.env_init = True
-            self.logger.stash_results(infos, name_override="baseline")
+
+            # We only stash the results if we're not doing HPO, or else the results from concurrent HPO would get
```

Comment:
I think it's because we pick the name of the folder based on the timestamp, and two `stash_results()` calls could happen to be at the same timestamp. I don't believe it relates to linking artifacts. One reason you might not faced this is that you might not have done SF0.01 tests, where `stash_results()` is called at a very high frequency. Is stashing results during HPO useful? If so, I can add support for that.

Reply:
I think we still want to be stashing results during HPO since it's also a potential debugging/information source. We could shift away from the timestamp and just use a counter instead which should then fix the unique-ness problem.